google/cloud/datastore/client.py

```diff
@@ -199,8 +199,22 @@ def _extended_lookup(
 
     results = []
 
+    transaction_id = None
+    new_transaction_options = None
+    if transaction is not None:
+        transaction_id = transaction.id
+        if (
+            transaction_id is None
```

Comment:
Is this check redundant? ie. if `transaction._status == transaction._INITIAL` is false then will `transaction_id is None` always be false? If so then we can remove this.

Reply:
I wouldn't expect the transaction to be in a state where the status is INITIAL and an id is set, but that's not enforced. So the question is what we want to do if we get in that state

- If we keep this check, it will fall back to its previous behaviour of just using the transaction id
- If we remove it, it will raise an exception, because it tries to set an id and create a new transaction in the same rpc

Thinking again, I think you might be right that it would be better to let it raise an exception in this situation. But it's not completely clear-cut, so I'll have another look at the context